wemake_python_styleguide/violations/best_practices.py

```diff
@@ -1851,3 +1852,34 @@ class ApproximateConstantViolation(ASTViolation):
 
     code = 447
     error_template = 'Found approximate constant: {0}'
+
+
+@final
+class MisrefactoredAssignmentViolation(ASTViolation):
+    """
+    Forbids to use mis refactored self assignment.
+
+    Reasoning:
+        self assignment need not have the same operand on the right hand side
+
+    Solution:
+        Raise violation if same operand is found on both sides
```

Comment:
`Refactor you code to use multiple self assignments or fix your code.`

Reply:
added